pkg/config/autodiscover/csv_info_test.go

```diff
@@ -57,3 +58,106 @@ func TestCSVGetAnnotationValue(t *testing.T) {
 	assert.Equal(t, []string{"OPERATOR_STATUS", "ANOTHER_TEST"}, val)
 	assert.Nil(t, err)
 }
+
+func TestAnnotationUnmarshalError(t *testing.T) {
+	testCases := []struct {
+		expectedString string
+	}{
+		{
+			expectedString: "error (this is an error) attempting to unmarshal value of annotation 'testKey' on CSV 'testnamespace/testname'",
+		},
+	}
+
+	for _, tc := range testCases {
+		csvr := CSVResource{
+			Metadata: struct {
+				Name        string            "json:\"name\""
+				Namespace   string            "json:\"namespace\""
+				Labels      map[string]string "json:\"labels\""
+				Annotations map[string]string "json:\"annotations\""
+			}{
+				Name:      "testname",
+				Namespace: "testnamespace",
+			},
+		}
+		assert.Equal(t, tc.expectedString, csvr.annotationUnmarshalError("testKey", errors.New("this is an error")).Error())
+	}
+}
+
+func TestGetCSVsByLabel(t *testing.T) {
+	testCases := []struct {
+		filename    string
+		label       string
+		value       string
+		expectedCSV string
+	}{
+		{
+			filename:    "csv_output.json",
+			expectedCSV: "etcdoperator.v0.9.4",
+			label:       "testLabel",
+			value:       "testValue",
+		},
+		{
+			filename:    "csv_output_nolabel.json",
+			expectedCSV: "",
+			label:       "testLabel",
+			value:       "testValue",
+		},
+	}
+
+	for _, tc := range testCases {
+		origFunc := executeOcGetAllCommand
+		executeOcGetAllCommand = func(resourceType, labelQuery string) string {
+			output, err := os.ReadFile(path.Join(filePath, tc.filename))
+			assert.Nil(t, err)
+			return string(output)
+		}
+
+		outputList, err := GetCSVsByLabel(tc.label, tc.value)
+		assert.Nil(t, err)
+		if len(outputList.Items) > 0 {
+			assert.Equal(t, tc.expectedCSV, outputList.Items[0].Metadata.Name)
+		}
+
+		executeOcGetAllCommand = origFunc
+	}
+}
+
+func TestGetCSVsByNamespace(t *testing.T) {
+	testCases := []struct {
+		filename    string
+		label       string
+		value       string
+		expectedCSV string
+	}{
+		{
+			filename:    "csv_output.json",
+			expectedCSV: "etcdoperator.v0.9.4",
+			label:       "testLabel",
+			value:       "testValue",
+		},
+		{
+			filename:    "csv_output_nolabel.json",
+			expectedCSV: "",
+			label:       "testLabel",
+			value:       "testValue",
+		},
+	}
+
+	for _, tc := range testCases {
+		origFunc := executeOcGetAllCommand
+		executeOcGetCommand = func(resourceType, labelQuery, namespace string) string {
+			output, err := os.ReadFile(path.Join(filePath, tc.filename))
+			assert.Nil(t, err)
+			return string(output)
+		}
+
+		outputList, err := GetCSVsByLabelByNamespace(tc.label, tc.value, "testnamespace")
+		assert.Nil(t, err)
+		if len(outputList.Items) > 0 {
```

Comment:
I believe that the number of parsed Items should also be checked with something like:
`assert.Equal(t, len(outputList.Items), tc.ExpectedItemsCount)`

Reply:
Good idea, I'll add an expected items count.

Reply:
Done.